tests/test.sh

```diff
@@ -854,6 +854,74 @@ zero_local_jobs_test()
     echo
 }
 
+environment_extraction_priority_test()
+{
+    echo Running environment_extraction_priority_test test.
+    reset_logs remote "$@"
+
+    mkdir "$testdir/new_test/" &> /dev/null
+
+    test_priorities=(10 12 15)
+    for compile_priority in ${test_priorities[@]}; do
+        kill_daemon remoteice1
+        start_iceccd remoteice1 -p 10246 -m 2
+        for time in $(seq 1 10); do
+            notready=
+            if ! kill  -0 ${remoteice1_pid}; then
+                echo Daemon remoteice1 start failure.
+                stop_ice 0
+                abort_tests
+            fi
+            if ! grep -q "Connected to scheduler" "$testdir/remoteice1.log"; then
+                # ensure log file flush
+                kill -HUP ${remoteice1_pid}
+                grep -q "Connected to scheduler" "$testdir/remoteice1.log" || notready=1
+            fi
+            if test -z "$notready"; then
+                break;
+            fi
+            sleep 1
+        done
+        if test -n "$notready"; then
+            echo Icecream not ready, aborting.
+            stop_ice 0
+            abort_tests
+        fi
+        sleep 5 # Give the scheduler time to get everything set up
+        flush_logs
+        ICECC_TEST_SOCKET="$testdir"/socket-localice ICECC_TEST_REMOTEBUILD=1 ICECC_PREFERRED_HOST=remoteice1 ICECC_DEBUG=debug ICECC_LOGFILE="$testdir"/icecc.log \
+            nice -n ${compile_priority} $valgrind "${icecc}" $GXX -Wall -Werror -g -c plain.cpp -o "$testdir/new_test/plain.o" 2>"$testdir"/stderr.remoteice &
+        compile_pid=$!
+        for wait_count in $(seq 1 100); do
```

Comment:
This looks like a race condition.  On a loaded machine tar might not start in the expected time, while on a fast machine tar might finish before the test gets here.

I don't know how to test this code... I don't think it is possible to do it with the current style of tests we have.  I think a unit test of some sort is needed. (we currently are not structured to unit test the code in that way)